Library/Homebrew/livecheck/strategy.rb

```diff
@@ -166,20 +166,55 @@ def from_url(url, livecheck_strategy: nil, regex_provided: false, block_provided
         end
       end
 
+      # Creates `curl` `--data` or `--json` arguments (for `POST` requests`
+      # from `livecheck` block `url` options.
+      sig { params(url_options: T::Hash[Symbol, T.untyped]).returns(T::Array[String]) }
+      def post_args(url_options)
+        type = if (data = url_options[:data].presence)
+          :data
+        elsif (data = url_options[:json].presence)
+          :json
+        end
+        return [] if !type || !data
+
+        case data
+        when Hash
+          if type == :json
+            require "json"
+            ["--#{type}", JSON.generate(data)]
+          else
+            require "uri"
+            ["--#{type}", URI.encode_www_form(data)]
+          end
+        else
+          ["--#{type}", data]
+        end
```

Comment:
`post_args` provides support for data that's a string (already in the expected format) or a hash (that will be converted to the expected format). This allows for some flexibility but I imagine we will standardize on one format in core/cask, so it may not be necessary.

Hashes may be easiest to work with and maintain, so I could drop string support if there's agreement. This would allow me to simplify the `post_args` implementation and general `POST` support documentation. [Though we'll probably never use it, one bonus of using hashes is that we can easily switch between data formats by simply changing `data:` to `json:`.]